lib/blocks.tsx

```diff
@@ -0,0 +1,95 @@
+import { createElement, ReactNode } from 'react';
+import { ContentBlock, ContentBlockAttribute } from '@/graphql/generated';
+import {
+	ClassicEditorBlock,
+	Heading,
+	BlockImage,
+	List,
+	Paragraph,
+	Quote,
+	Table,
+	UnsupportedBlock,
+} from '@/components/Blocks/index';
+
+/**
+ * Map an array of ContentBlock attributes to an object that can used like props.
+ */
+function mapAttributesToProps ( attributes: ContentBlockAttribute[] ): { [ key: string ]: string; } {
+	return attributes.reduce( ( acc, { name, value } ) => {
+		// Drop attributes without a name or value.
+		if ( ! name || ! value ) {
+			return acc;
+		}
+
+		// Values are always strings, so we could cast some "special" strings to
+		// their presumed types. For example, we could convert "false" to false,
+		// "null" to null, etc.
+		//
+		// This could cause unexpected issues, so we will leave that exercise for
+		// those that find a need for it.
+
+		return Object.assign( acc, { [ name ]: value } );
+	}, {} );
+}
+
+type BlocksToComponentsProps = Record<string, ReactNode>;
+
+/**
+ * Return an object with respected key-values based on Gutenberg Blocks
+ * https://gogutenberg.com/blocks/
+ */
+function mapBlockNamesToComponents ( override?: BlocksToComponentsProps ): Record<string, ReactNode> {
```

Comment:
I don't think I really see the usefulness of this as a function, especially how we call it during render. The behavior never changes so a function seems overly complicated. All we really want is to provide a way to override the default mapping. So we could just provide a `blockMapOverrides` prop on `PostContent`. Same result, and it opens the door to sealing `PostContent` and providing it via a library. 

```
import PostContent from '@/lib/components'; // or maybe an NPM package eventually
import MyCustomHeader from 'my-design-system';

export default function Post( props: Props ) {
  return (
    <main>
      <h1>{props.title}</h1>
      <PostContent
        blockMapOverrides={ {
          'core/heading': MyCustomHeader,
        } }
        blocks={props.blocks}
      />
    </main>
  );
}
```

It also would allow users to provide different mappings in different contexts. I could provide a different mapping when rendering a page vs a post:

```
import PostContent from '@/lib/components'; // or maybe an NPM package eventually
import MyPageHeader from 'my-design-system';

export default function Page( props: Props ) {
  return (
    <main>
      <h1>Page: {props.title}</h1>
      <PostContent
        blockMapOverrides={ {
          'core/heading': MyPageHeader,
        } }
        blocks={props.blocks}
      />
    </main>
  );
}
```

Reply:
@chriszarate That's a great idea, updates are done :)